### DISCUSSION TITLE
Display Sway mode?

### DISCUSSION THREAD

**Discussion by reidevries** [nwg-piotr/nwg-shell#174] [Ideas]
Is there any way currently to display the current Sway mode? I use modes in my custom Sway+Waybar config extensively, and usually they would be displayed on the panel. But in nwg-shell default config when I enter resize mode, it doesn't display the mode anywhere on the screen and it's impossible to tell whether I'm in resize mode. 

If not, a panel item for "Sway mode" would be really useful, all it has to do is display text when Sway is in a particular mode.

**Comment by nwg-piotr** [comment]
This could be done as an executor, but quite difficult, I must admit. I'll take a closer look and try to write a module.

**Comment by nwg-piotr** [comment]
Seems to work and look not bad:

![image](https://github.com/nwg-piotr/nwg-shell/assets/20579136/5fbedcb9-1839-460d-9093-e31565f86199)

I still need to add config.

**Comment by nwg-piotr** [comment]
![image](https://github.com/nwg-piotr/nwg-shell/assets/20579136/fb63b988-76b8-47ff-bfbd-fe2d235eb757)

**Comment by nwg-piotr** [comment]
About to release as nwg-panel v0.9.12. The SwayMode module on the top right, while "resize" mode selected. You need to add the module to your panel.

![2023-08-23-002759_sway_screenshot](https://github.com/nwg-piotr/nwg-shell/assets/20579136/7c9d5dc3-7bc9-4a14-aa5d-e47fcc0e2144)

**Comment by nwg-piotr** [comment]
Released: https://github.com/nwg-piotr/nwg-panel/releases/tag/v0.9.12

**Comment by reidevries** [comment]
Wow nice! Looks perfect! Thanks :D

**Reply by nwg-piotr** [reply]
You're welcome. :)
